### PR TITLE
[MOB-3622] track in-app close events

### PR DESCRIPTION
**Pull request by evangreer91**
## 🔹 JIRA Ticket(s) if any

* [MOB-3622](https://iterable.atlassian.net/browse/MOB-3622)

## ✏️ Description

This pull request includes adding in-app close events to the handleHTMLmessage function when the dismiss link is clicked and to the onPress prop of the return to inbox button.

### CODE REVIEW

File: ts/IterableInboxMessageDisplay.tsx [hunk 1]
```diff
@@ -74,13 +74,14 @@ const IterableInboxMessageDisplay = ({
...
 
-      Iterable.trackInAppClick(rowViewModel.inAppMessage, IterableInAppLocation.inApp, url)
+      Iterable.trackInAppClick(rowViewModel.inAppMessage, IterableInAppLocation.inbox, url)
       
       if(url === 'iterable://delete') {
          deleteRow(rowViewModel.inAppMessage.messageId)
          returnToInbox()
       } else if(url === 'iterable://dismiss') {
          returnToInbox()
+         Iterable.trackInAppClose(rowViewModel.inAppMessage, IterableInAppLocation.inbox, IterableInAppCloseSource.link, 'iterable://dismiss')
```
Comment (roninopf) [inline ts/IterableInboxMessageDisplay.tsx:84]: Whoops. I think our `trackInAppClose` isn't complete, the `clickedUrl` field here should just be `null`, so we should build that support in.

File: ts/IterableInboxMessageDisplay.tsx [hunk 2]
```diff
@@ -101,7 +102,11 @@ const IterableInboxMessageDisplay = ({
    return(
       <View style={updatedMessageDisplayContainer}>
          <View style={returnButtonContainer}>
-            <TouchableWithoutFeedback onPress={() => returnToInbox()}>
+            <TouchableWithoutFeedback 
+               onPress={() => {
+                  returnToInbox()
+                  Iterable.trackInAppClose(rowViewModel.inAppMessage, IterableInAppLocation.inbox, IterableInAppCloseSource.back, 'iterable://dismiss')
+               }}>
```
Comment (roninopf) [inline ts/IterableInboxMessageDisplay.tsx:109]: Same to here, we should allow `clickedUrl` to be `null`.

File: ts/IterableInboxMessageDisplay.tsx (hunk 1 above)
Comment (evangreer91) [reply ts/IterableInboxMessageDisplay.tsx:84]: I will make the update.

File: ts/IterableInboxMessageDisplay.tsx (hunk 2 above)
Comment (evangreer91) [reply ts/IterableInboxMessageDisplay.tsx:109]: I will make the update.

File: ts/IterableInboxMessageDisplay.tsx (hunk 1 above)
Comment (evangreer91) [reply ts/IterableInboxMessageDisplay.tsx]: I added undefined for the clickedUrl field

File: ts/IterableInboxMessageDisplay.tsx (hunk 2 above)
Comment (evangreer91) [reply ts/IterableInboxMessageDisplay.tsx:109]: I added undefined for the clickedUrl field.

**Review by roninopf — APPROVED** [review]

**Pull request merged**